### ISSUE TITLE
Progression

### ISSUE THREAD

**Issue by GreatestFool** [GreatestFool/crystialis#27]
A consideration of whether or not to have progression. Or rather, what form it will take, some manner of progression will obviously be in place. Like the boss dimension mentioned in #15.

Currently very early stage, but some form of expert pack without the grind-y-ness?

Mix tech and magic, where tech unlocks tech and magic unlocks magic until endgame, ala Nuts and Bolts Torqued?

---

It seems like most of the threads I've checked out regarding expert packs find the best ones to be the ones that don't meaninglessly time gate stuff. But rather make things [hard to make at first](https://www.reddit.com/r/feedthebeast/comments/9hzvpt/what_makes_a_good_expert_pack/), but once you get further into the pack, you find other methods in which to achieve the same thing faster or more efficiently, allowing for the player to feel like their effort is paying off. It's a fine line to thread though, [make it to apparent, and a contraption a player made can feel invalidated](https://www.reddit.com/r/feedthebeast/comments/9hzvpt/what_makes_a_good_expert_pack/e6fu1ri/).

To play off of that, there's also not meaninglessly invalidating things the moment you get it. [This comment puts it rather well](https://www.reddit.com/r/feedthebeast/comments/9hzvpt/what_makes_a_good_expert_pack/e6g1hxt/), I think. Yet, as the other commenter in the thread mentions, you also have to be careful of letting stuff stay relevant for _too_ long, e.g. a horse-shoe grinder still being the best way to make coal in SevTech: Ages' spaceage.

[This thread](https://www.reddit.com/r/feedthebeast/comments/6f5ulr/to_modpack_makers_how_i_believe_expert_packs_can/) also puts the difference between knowledge of mechanics vs not knowing them into a nice manner: _I believe that single blocks that do everything for you like quarries should requiring grinding to achieve but setups similar to [Direwolf20's Frame Quarry](https://www.youtube.com/watch?v=3A1ZTN3KLyE) shouldn't. These builds would instead be gated behind knowledge rather than resources allowing players who learn the game mechanics to get things such as quarries up early on but also allow those who are fine with grinding to get a quarry._

[Create](https://www.curseforge.com/minecraft/mc-mods/create) probably is able to serve as this, in regards to Direwolf20's Frame Quarry, likewise, Crossroads MC probably also fits.

And also to tightly integrate mods, most mention Gregtech New Horizons for this, I'll have to check it out myself- but, on the top of my head, I could imagine one way to do this is to make mods require resources from each other _however_, without any way to automatically spam that resource aside from the tools the mod gives you, forcing you to actually go through the mod to automate it or something, or finding some other creative solution?

There's also being open ended, in the sense of allowing for multiple potential ways (tons to be crude) to solve a particular problem. Essentially, not artificially locking a player in to one mod or one block, it is embracing the content overlap and exemplifying it in a way that's fun to play without invalidating each other.

Basically, the four rules are:

1. Some resource or recipe gating, such that it is hard and satisfying but not tedium or a time gate
2. Intricacy over complexity, essentially simple functions that can be used in multiple scenarios, over complex single-field functions
3. Cross-mod integration, without trivializing their respective developmental trees
4. Open-endedness, allowing the player to be creative and think up solutions to problems in their own way

Will investigate more.

**Comment by GreatestFool** [comment]
Hmm, now that I think about this. For expert-type packs, which this seems to be becoming some pseudo-version of, something like Iron/Metal Furnaces or whatever is probably a better idea to _not_ include, because you invalidate much cooler and potentially better, more fun or whatever systems, simply because you can just use that one single block that does it super duper fast.

[This comment thread](https://www.reddit.com/r/feedthebeast/comments/6f5ulr/to_modpack_makers_how_i_believe_expert_packs_can/difokdt/) that made me think of it.

**Comment by GreatestFool** [comment]
[This comment](https://www.reddit.com/r/feedthebeast/comments/6f5ulr/to_modpack_makers_how_i_believe_expert_packs_can/digt96f/) also makes a great point in terms to increased recipe costs to something like Veinminer, Ore Exavator, FTB Ultimine or similarly.

**Comment by GreatestFool** [comment]
One potential way to add resource increase progression is to start out vanilla costs, them progressively get more and more expensive per 'age' and thus incentivizing automating through increased difficulty _over time_ aside from right out of the gate.

Thus, in regards to #13, wood and/or cobble tools should probably be left alone?

---

There's also [this comment](https://www.reddit.com/r/feedthebeast/comments/6f5ulr/to_modpack_makers_how_i_believe_expert_packs_can/digtmj2/), specifically _I don't disagree that Kappa mode is grindy but there are still ways to alleviate or get around the grind; furnaces, too, 8 cobblestone gets you a slab furnace, you can't upgrade it but you can still use it to smelt things._

That furnace idea might work here too.

**Comment by GreatestFool** [comment]
Very rough progression stuff thingy:

Early game == overworld
Middle game == the nether
End game == the end

To gain the keys of access, you likewise have to defeat a good few bosses. Not as many for the nether, but there are twelve keys needed for the end, each carried by a boss. To avoid pointless loot, this is given through kill quests.
